Zeeguu Reader/Zeeguu Reader/LoginTableViewController.swift

```diff
@@ -89,11 +89,15 @@ class LoginTableViewController: ZGTableViewController {
 		
 		if let em = email, pw = password {
 			ZeeguuAPI.sharedAPI().loginWithEmail(em, password: pw) { (success) -> Void in
-				if (success) {
-					self.dismissViewControllerAnimated(true, completion: nil)
-				} else {
-					Utils.showOKAlertWithTitle("LOGIN_FAILED".localized, message: "LOGIN_FAILED_MESSAGE".localized, okAction: nil)
-				}
+				dispatch_async(dispatch_get_main_queue(), {
```

Comment:
i'm not familiar with dispatch_async but it looks like you're starting a new thread. shouldn't you do some computation in that thread?